### PR TITLE
Implemented test for drop of all subsequent contributions

### LINKED ISSUE #122: Write test to ensure dropped contributor removes all subsequent contributions
If a contributor is dropped, all subsequent contributions from that party must also be dropped.

### PR DESCRIPTION
**Pull request by ibaryshnikov**
Closes #122 

I'm not totally sure about the params, but I wanted to have a small number of tasks in this test in order to check them concisely.
```
1, /* power */
2, /* batch_size */
2, /* chunk_size */
```

### CODE REVIEW

File: phase1-coordinator/src/tests.rs [hunk 1]
```diff
@@ -1033,6 +1035,72 @@ fn coordinator_drop_contributor_clear_locks_test() -> anyhow::Result<()> {
...
+        };
+        assert_eq!(contributor_info.assigned_tasks().len(), 0);
+        assert_eq!(contributor_info.completed_tasks(), &expected_tasks);
+    }
+
+    let locators = coordinator.drop_participant(&contributor1)?;
+    assert_eq!(2, locators.len());
+
+    for (contributor, contributor_info) in coordinator.current_contributors() {
+        if contributor == contributor2 {
```
Comment (kellpossible) [inline phase1-coordinator/src/tests.rs:1092]: doing a `match` here might be a bit more explicit for what you're asserting

```rust
match contributor {
	contributor1 => {
		assert_eq!(contributor_info.completed_tasks().len(), 0);
    	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 1), (1, 2)]));
	}
	contributor2 => {
		assert_eq!(contributor_info.completed_tasks(), &make_tasks(&[(1, 1)]));
	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 2)]));
	}
	unexpected  => {
		panic!("current_contributors() contains an unexpected contributor: {:?}", unexpected)
	}
}
```

File: phase1-coordinator/src/tests.rs [hunk 2]
```diff
@@ -1033,6 +1035,72 @@ fn coordinator_drop_contributor_clear_locks_test() -> anyhow::Result<()> {
...
+
+    for _ in 0..number_of_chunks {
+        coordinator.contribute(&contributor1, &contributor_signing_key1, &seed1)?;
+        coordinator.contribute(&contributor2, &contributor_signing_key2, &seed2)?;
+        coordinator.verify(&verifier, &verifier_signing_key)?;
+        coordinator.verify(&verifier, &verifier_signing_key)?;
+    }
+
+    for (contributor, contributor_info) in coordinator.current_contributors() {
+        let expected_tasks = if contributor == contributor1 {
```
Comment (kellpossible) [inline phase1-coordinator/src/tests.rs:1079]: doing a `match` here might be a bit more explicit for what you're asserting (see same comment below for expanded explanation).

File: phase1-coordinator/src/tests.rs (hunk 1 above)
Comment (ibaryshnikov) [reply phase1-coordinator/src/tests.rs:1092]: That's very interesting, didn't know it was possible! I've tried it in playground and it doesn't work for some reason, can you check it?
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e7c3af9c3e1231c28364dc89067e653d

Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1092]: @ibaryshnikov oooh okay, yeah you're right, I guess numbers/patterns need to be const in the match! Not sure how my brain mixed that one up. It looks like there [might still be a way to do it](https://stackoverflow.com/questions/55804275/are-there-traditional-style-switch-statements-in-rust).  https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f9ee703a8cede19e064b4228b08bbeeb

```rust
match contributor {
	c if c == contributor1 => {
		assert_eq!(contributor_info.completed_tasks().len(), 0);
    	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 1), (1, 2)]));
	}
	c if c == contributor2 => {
		assert_eq!(contributor_info.completed_tasks(), &make_tasks(&[(1, 1)]));
	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 2)]));
	}
	unexpected  => {
		panic!("current_contributors() contains an unexpected contributor: {:?}", unexpected)
	}
}
```

Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1092]: I just realised that the `else` block might not be matching `contributor1` but may also be matching the coordinator contributor, which seems to be a thing...

Comment (ibaryshnikov) [reply phase1-coordinator/src/tests.rs:1098]: Your comments helped me to improve the documentation and add a check for unexpected contributor, thanks!

**Review by howardwu — APPROVED** [review]

**Pull request merged**